charts/promtail/templates/hpa.yaml

```diff
@@ -1,5 +1,9 @@
 {{- if and .Values.deployment.enabled .Values.deployment.autoscaling.enabled }}
-apiVersion: autoscaling/v2beta1
+apiVersion: {{ if or (.Capabilities.APIVersions.Has "autoscaling/v2/HorizontalPodAutoscaler") (semverCompare ">=1.23" .Capabilities.KubeVersion.Version) -}}
+autoscaling/v2
+{{- else -}}
+autoscaling/v2beta2
+{{- end }}
```

Comment:
```suggestion
apiVersion: autoscaling/v2{{ or (.Capabilities.APIVersions.Has "autoscaling/v2/HorizontalPodAutoscaler") (semverCompare ">=1.23" .Capabilities.KubeVersion.Version) | ternary "" "beta2" }}
```

Reply:
I considered this. It's so much more difficult to read on a single line.